flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableSource.java

```diff
@@ -261,6 +279,93 @@ public DynamicTableSource copy() {
         return source;
     }
 
+    @Override
+    public TableStats reportStatistics() {
+        try {
+            // only support BOUNDED source
+            if (isStreamingSource()) {
+                return TableStats.UNKNOWN;
+            }
+            if (flinkConf.get(HiveOptions.SOURCE_REPORT_STATISTICS)
+                    != FileSystemConnectorOptions.FileStatisticsType.ALL) {
+                return TableStats.UNKNOWN;
+            }
+
+            HiveSourceBuilder sourceBuilder =
+                    new HiveSourceBuilder(jobConf, flinkConf, tablePath, hiveVersion, catalogTable)
+                            .setProjectedFields(projectedFields)
+                            .setLimit(limit);
```

Comment:
we should consider how to handle the case after limit push and filter push down

Reply:
> we should consider how to handle the case after limit push and filter push down

Now, hive source don't support filter push down. For limit push down, `PushLimitIntoTableSourceScanRule` happened after `FlinkRecomputeStatisticsProgram`, and `PushLimitIntoTableSourceScanRule` can re-compute  the new row count. So, I think there is no need to add re-compute stats logic in HiveTableSource.

Reply:
> 

Adding limit push down logic in HiveTableSource for reporting stats. But, this not work now because of optimizing order.